src/java/com/sarality/app/view/BaseViewInitializer.java

```diff
@@ -26,13 +26,13 @@
  *
  * @param <V> Type of view being initialized.
  * @param <T> Type of data needed to initialize the view.
- *
  * @author abhideep@ (Abhideep Singh)
  */
 public abstract class BaseViewInitializer<V extends View, T> implements ViewInitializer<V, T>, DataConsumer<T> {
 
   private static final Logger logger = LoggerFactory.getLogger(BaseViewInitializer.class);
 
+  private static int loader = 0;
```

Comment:
I had given it some thought in terms of supporting multiple Loader and I think the better approach might be to define the static final int in each Activity and Fragment and then passing it into the Initializer. Auto increment is dangerous since it will simply keep creating new loaders rather than reusing loaders.